lib/traject/bibtex_config.rb

```diff
@@ -34,18 +34,59 @@
   accumulator << record.author.to_s.presence
 }
 
+to_field 'editor_ssim', lambda { |record, accumulator, _context|
+  accumulator << record.editor.to_s.presence if record.respond_to?(:editor)
+}
+
+to_field 'book_title_ssim', lambda { |record, accumulator, _context|
+  accumulator << record.booktitle.to_s.presence if record.respond_to?(:booktitle)
+}
+
 to_field 'pub_display', lambda { |record, accumulator, _context|
   accumulator << record.journal.to_s.presence if record.respond_to?(:journal)
+  accumulator << record.publisher.to_s.presence if record.respond_to?(:publisher)
+}
+
+to_field 'pub_year_w_approx_isi', lambda { |record, accumulator, _context|
+  accumulator << record.year.to_s.presence if record.respond_to?(:year)
+}
+
+to_field 'location_ssi', lambda { |record, accumulator, _context|
+  accumulator << record.address.to_s.presence if record.respond_to?(:address)
+}
+
+to_field 'university_ssim', lambda { |record, accumulator, _context|
+  accumulator << record.school.to_s.presence if record.respond_to?(:school)
+}
+
+to_field 'edition_ssm', lambda { |record, accumulator, _context|
+  accumulator << record.edition.to_s.presence if record.respond_to?(:edition)
+}
+
+to_field 'series_ssi', lambda { |record, accumulator, _context|
+  accumulator << record.series.to_s.presence if record.respond_to?(:series)
+}
+
+to_field 'thesis_type_ssm', lambda { |record, accumulator, _context|
+  accumulator << record.fields[:type].to_s.presence if record.fields.include?(:type)
```

Comment:
I have a more verbose variation here because of a conflict between .BibTeX and Traject. `phdthesis` items have a `type` field that indicates the degree level (e.g. B.Litt). Traject normally points to `record.type` points to the reference type (e.g. `phdthesis`, `incollection`, etc.)